source/encoder.c

```diff
@@ -345,17 +348,19 @@ CBOREncoder_write(CBOREncoderObject *self, PyObject *data)
     Py_RETURN_NONE;
 }
 
-
 static int
-encode_length(CBOREncoderObject *self, const uint8_t major_tag,
-              const uint64_t length)
+encode_length_possibly_indefinite(CBOREncoderObject *self, const uint8_t major_tag,
```

Comment:
Can you explain why you had to split this function into two parts?

Reply:
1. I wanted to reduce diff by keeping rest of the encode_length calls the same (not having to add the extra parameter everywhere)
2. I felt that the bool parameter would be somewhat cryptic if there was not some addition to the function name, so I created `encode_length_possibly_indefinite`. Most of the use cases don't care about indefinite encoding, so it didn't make sense to drag this function there - so we have two now.

Reply:
I took a better look at this, and it doesn't make a whole lot of sense. Since `uint64_t` cannot possibly be -1 since it's unsigned. The actual value would then become a very large integer (18446744073709551615) instead, yes? While I doubt anyone will really try to actually encode such enormous structures, it does raise other questions, like if you felt this was an acceptable sentinel value, why then did you have to split `encode_length()` instead of just using -1 as the sentinel for indefinite length? And why can't `encode_length()` just directly look at `self->indefinite_containers` and if it's true, ignore the length parameter?

Reply:
encode_length cannot directly use indefinite_containers, because it is also used on other places outside of encoding array/map length.